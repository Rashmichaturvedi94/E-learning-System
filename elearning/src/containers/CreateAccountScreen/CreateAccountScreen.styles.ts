import styled from 'styled-components/native';
import { StyleSheet, SafeAreaView, TextInput, Text, TouchableHighlight } from 'react-native';
import { CreateAccountScreenProps } from './CreateAccountScreen.interface';


export const Container = styled(SafeAreaView)`
  flex: 1;
  justify-content: center;
  margin: 0px 16px;
`;

export const TitleText = styled(Text)`
  text-align: center;
  font-size: 30px;
  font-weight: bold;
  margin-bottom: 100px;
`;

export const SignInText = styled(Text)`
  text-align: center;
  font-size: 14px;
  margin-top: 100px;
`;

export const AggrementText = styled(Text)`
  font-size: 14px;
  margin-top: 30px;
`;

export const SignInLink= styled(Text)`
  text-align: center;
  font-size: 14px;
  font-weight: bold;
  color: #0717f7;
`;
export const Input = styled(TextInput)`
  height: 40px;
  margin-top: 20px;
  border-bottom-color: #b0adac;
  border-bottom-width: 0.5px;
  border-color: white;
`;

export const SaveButton = styled(TouchableHighlight)`
  height: 50px;
  border-radius: 10px;
  background-color: #1b8ffa;
  margin-top: 100px;
  justify-content: center;
  align-items: center;
`;

export const SaveText = styled(Text)`
  color: #FFFFFF;
`;
