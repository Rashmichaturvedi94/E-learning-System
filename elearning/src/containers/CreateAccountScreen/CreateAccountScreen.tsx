import React from 'react';
import { View, Alert } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import {
  Container,
  Input,
  TitleText,
  SaveButton,
  SaveText,
  SignInText,
  SignInLink,
  AggrementText,
} from './CreateAccountScreen.styles';

export const CreateAccountScreen = () => {
  const navigation = useNavigation();

  return (
    <Container>
      <View>
        <TitleText>Create Acount</TitleText>
        <Input placeholder="Name" />
        <Input placeholder="Email" />
        <Input placeholder="password" />
        <AggrementText>I agree to terms and conditions</AggrementText>
        <SaveButton onPress={() => Alert.alert('Simple Button pressed')}>
          <SaveText>Create Account</SaveText>
        </SaveButton>
        <SignInText>
          Already have account?
          <SignInLink
            onPress={() => {
              navigation.navigate('Login');
            }}
          >
            Sign In
          </SignInLink>
        </SignInText>
      </View>
    </Container>
  );
};
