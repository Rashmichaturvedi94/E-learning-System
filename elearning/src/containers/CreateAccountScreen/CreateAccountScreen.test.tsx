import React from 'react';
import { render } from '@testing-library/react-native';
import { CreateAccountScreen } from './CreateAccountScreen';

describe('containers/CreateAccountScreen', () => {
  it('should render', () => {
    const component = render(<CreateAccountScreen />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
