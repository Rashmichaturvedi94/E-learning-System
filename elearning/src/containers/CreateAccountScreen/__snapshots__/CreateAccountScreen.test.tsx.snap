// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/CreateAccountScreen should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Array [
                          Object {
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "justifyContent": "center",
                            "marginBottom": 0,
                            "marginLeft": 16,
                            "marginRight": 16,
                            "marginTop": 0,
                          },
                        ]
                      }
                    >
                      <View>
                        <Text
                          style={
                            Array [
                              Object {
                                "fontSize": 30,
                                "fontWeight": "bold",
                                "marginBottom": 100,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Create Acount
                        </Text>
                        <TextInput
                          allowFontScaling={true}
                          maxLength={50}
                          onChangeText={[Function]}
                          placeholder="Name"
                          rejectResponderTermination={true}
                          style={
                            Array [
                              Object {
                                "borderBottomColor": "#b0adac",
                                "borderBottomWidth": 0.5,
                                "borderColor": "white",
                                "height": 40,
                                "marginTop": 20,
                              },
                            ]
                          }
                          underlineColorAndroid="transparent"
                        />
                        <TextInput
                          allowFontScaling={true}
                          autoCapitalize="none"
                          autoCompleteType="email"
                          keyboardType="email-address"
                          maxLength={50}
                          onChangeText={[Function]}
                          placeholder="Email"
                          rejectResponderTermination={true}
                          style={
                            Array [
                              Object {
                                "borderBottomColor": "#b0adac",
                                "borderBottomWidth": 0.5,
                                "borderColor": "white",
                                "height": 40,
                                "marginTop": 20,
                              },
                            ]
                          }
                          underlineColorAndroid="transparent"
                        />
                        <TextInput
                          allowFontScaling={true}
                          maxLength={20}
                          onChangeText={[Function]}
                          placeholder="password"
                          rejectResponderTermination={true}
                          secureTextEntry={true}
                          style={
                            Array [
                              Object {
                                "borderBottomColor": "#b0adac",
                                "borderBottomWidth": 0.5,
                                "borderColor": "white",
                                "height": 40,
                                "marginTop": 20,
                              },
                            ]
                          }
                          underlineColorAndroid="transparent"
                        />
                        <View
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                              },
                            ]
                          }
                        >
                          <View
                            accessibilityRole="checkbox"
                            accessibilityState={
                              Object {
                                "checked": false,
                              }
                            }
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "backgroundColor": "rgba(0,0,0,0)",
                                "borderColor": "#ededed",
                                "borderRadius": 3,
                                "borderWidth": 0,
                                "left": -16,
                                "margin": 5,
                                "marginLeft": 10,
                                "marginRight": 10,
                                "opacity": 1,
                                "padding": 10,
                                "top": 16,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#bfbfbf",
                                      "fontSize": 24,
                                    },
                                    Object {
                                      "minWidth": 24,
                                    },
                                    Object {
                                      "fontFamily": "FontAwesome",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                              >
                                
                              </Text>
                              <Text
                                style={
                                  Object {
                                    "color": "#43484d",
                                    "fontWeight": "400",
                                    "marginLeft": 10,
                                    "marginRight": 10,
                                  }
                                }
                                testID="checkboxTitle"
                              >
                                I agree to
                              </Text>
                            </View>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#007aff",
                                    "marginLeft": -42,
                                    "marginTop": 35,
                                    "textAlign": "left",
                                  },
                                ]
                              }
                            >
                              terms and conditions
                            </Text>
                          </View>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#1b8ffa",
                                "borderRadius": 10,
                                "height": 50,
                                "justifyContent": "center",
                                "marginTop": 100,
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#ffffff",
                                },
                              ]
                            }
                          >
                            Create Account
                          </Text>
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "alignSelf": "center",
                                "flexDirection": "row",
                                "marginTop": 100,
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "fontSize": 14,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            Already have account?
                          </Text>
                          <Text
                            onPress={[Function]}
                            style={
                              Array [
                                Object {
                                  "color": "#007aff",
                                  "fontSize": 14,
                                  "fontWeight": "bold",
                                  "marginLeft": 4,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          >
                            Sign In
                          </Text>
                        </View>
                      </View>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "height": 64,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
